rclcpp_lifecycle/include/rclcpp_lifecycle/state.hpp

```diff
@@ -49,8 +61,19 @@ class State
   label() const;
 
 protected:
+  rcutils_allocator_t allocator_;
+
   bool owns_rcl_state_handle_;
+
   const rcl_lifecycle_state_t * state_handle_;
+
+  RCLCPP_LIFECYCLE_PUBLIC
+  friend void
+  free_resources(State & instance);
+
+  RCLCPP_LIFECYCLE_PUBLIC
+  friend void
+  copy_from(State & lhs, const State & rhs);
```

Comment:
Both should still either not be member functions or operate on `this` and not receive `instance` / `lhs`. `copy_from` sounds a lot like `assign` to me. And `free_resources` like `reset`.